pkg/waifu_vault_api.go

```diff
@@ -15,16 +16,20 @@ import (
 
 const baseUrl = "https://waifuvault.moe"
 
-var client = &http.Client{}
+type api struct {
+	client http.Client
+}
 
-type Api struct {
+func NewWaifuvaltApi(client http.Client) mod.Waifuvalt {
+	return &api{
+		client: client,
```

Comment:
Ok, i can see point of customizable client there. 

But maybe, you wanna implement `options` pattern there, so user might omit passing client if he no intereseted in,  and instead we can create default one if it's not passed.

https://davidbacisin.com/writing/golang-options-pattern

Reply:
Yea, I use this a lot, but for this one option, I think it's OK not to have it